src/crypto.ts

```diff
@@ -53,3 +73,38 @@ function encodeString(str: string): Uint8Array {
   encoded.set(stringToUint8Array(str), 8);
   return encoded;
 }
+
+export function deriveChildSeed(masterSeed: string, seed: string): string {
+  return HashAll(masterSeed, seed).toString();
+}
+
+/**
+ * Generates a master key pair and seed.
+ * @param [length=64] - The number of random bytes for the seed. Note that the string seed will be converted to hex representation, making it twice this length.
+ */
+export function generateKeyPairAndSeed(length = 64): { publicKey: PublicKey; privateKey: SecretKey; seed: string } {
+  const seed = makeSeed(length);
+  return { ...keyPairFromSeed(seed), seed };
+}
+
+/**
+ * Generates a public and private key from a provided, secure seed.
+ * @param seed - A secure seed.
+ */
+export function keyPairFromSeed(seed: string): { publicKey: PublicKey; privateKey: SecretKey } {
+  // Get a 32-byte seed.
+  seed = pkcs5.pbkdf2(seed, "", 1000, 32, md.sha256.create());
+  return pki.ed25519.generateKeyPair({ seed });
+}
+
+function makeSeed(length: number): string {
+  if (isBrowser) {
+    const array = new Uint8Array(length);
+    // Use built-in cryptographically-secure random number generator.
+    window.crypto.getRandomValues(array);
+    return Buffer.from(array).toString("hex");
+  } else {
+    const array = crypto.randomBytes(length);
+    return Buffer.from(array).toString("hex");
+  }
```

Comment:
You won't need all these checks if you'd use https://www.npmjs.com/package/randombytes

Reply:
> 
> 
> You won't need all these checks if you'd use https://www.npmjs.com/package/randombytes
Browsers and Node also has built-in crypto module, I think it iis better than using an additional dependency. Browsers also has built in window.crypto

Reply:
from https://www.npmjs.com/package/randombytes

> randombytes from node that works in the browser. In node you just get crypto.randomBytes, but in the browser it uses .crypto/msCrypto.getRandomValues